src/ui/mainwindow.cpp

```diff
@@ -99,9 +101,9 @@ MainWindow::MainWindow(Core::Application *app, QWidget *parent) :
     // initialise ui
     ui->setupUi(this);
 
-    QShortcut *focusSearch = new QShortcut(QKeySequence(QStringLiteral("Ctrl+K")), this);
-    focusSearch->setContext(Qt::ApplicationShortcut);
-    connect(focusSearch, &QShortcut::activated,
+    m_focusSearch = std::unique_ptr<QShortcut>(new QShortcut(QKeySequence(QStringLiteral("Ctrl+K")), this));
```

Comment:
`QShortcut` is `QObject`, it is deleted when by its parent. Why `std::unique_ptr`?